File: app/src/main/java/com/github/dedis/student20_pop/model/network/layer/base/Catchup.java
package com.github.dedis.student20_pop.model.network.layer.base;

/**
 * Query to catchup on missed messages
 */
public final class Catchup extends Query {

    /**
     * Constructor for a Catchup
     *
     * @param channel name of the channel
     * @param id request ID of the catchup
     */
    public Catchup(String channel, int id) {
        super(channel, id);
    }

    @Override
    public String getMethod() {
        return Method.CATCHUP.getMethod();
    }
}

File: be1-go/message/query/method/catchup.go
package method

import "popstellar/message/query"

// Catchup define a JSON RPC catchup message
type Catchup struct {
	query.Base

	ID int `json:"id"`

	Params CatchupParams `json:"params"`
}

type CatchupParams struct {
	Channel string `json:"channel"`
}
